torch/distributions/bernoulli.py

```diff
@@ -41,7 +42,12 @@ class Bernoulli(ExponentialFamily):
     has_enumerate_support = True
     _mean_carrier_measure = 0
 
-    def __init__(self, probs=None, logits=None, validate_args=None):
+    def __init__(
+        self,
+        probs: Optional[Union[Tensor, Number]] = None,
```

Comment:
Actually, I think all of these should be replaced by `Union[Tensor, float]`, annotating as `Number` is aweful, because it is incompatible with `mypy`:
https://mypy-play.net/?mypy=latest&python=3.12&gist=87f5f362040b0d56bca3d3468d0224f9

Reply:
I used `Number` because that is what the docstring says. Does `torch` try to support `numpy` scalars, or why is `Number` used instead of simple `float`?

Reply:
@Skylion007 If one wants to be generic, one could use `typing.SupportsFloat` which would support 3rd party float types.

Reply:
See #144788

Reply:
I changed it to adding a single `assert precision_matrix is not None` as in the other thread.

Reply:
Fixed by #145086